topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -137,45 +143,133 @@ We don't want to overload our training VMs trying to run real tools, so to demon
+>           tools:
+>             - class: local # these special tools that aren't parameterized for remote execution - expression tools, upload, etc
+>               environment: local_env
+>    @@ -117,6 +98,10 @@ galaxy_config:
+>               - job-handlers
+>               - workflow-schedulers
+>
+>    +galaxy_config_files:
+>    +  - src: files/galaxy/config/tpv_rules_local.yml
+>    +    dest: "{{ galaxy_config_dir }}/tpv_rules_local.yml"
+>    +
+>     galaxy_config_templates:
+>       - src: templates/galaxy/config/container_resolvers_conf.yml.j2
+>         dest: "{{ galaxy_config.galaxy.containers_resolvers_config_file }}"
 >    {% endraw %}
 >    ```
->    {: data-commit="Configure testing tool in job conf"}
+>    {: data-commit="Add TPV to job config"}
+>
+>  Note that we set the default execution environment to the tpv_dispatcher, added the tpv_dispatcher itself as a dynamic runner, and removed all other destinations.
+>  Adding TPV as a runner will cause Galaxy to automatically install the `total-perspective-vortex` package on startup as a conditional dependency.
+>  Finally, we added a new config file named `tpv_rules_local.yml`, which we will create next.
+>
+> 2. Create a new file named `tpv_rules_local.yml` in the "files" folder of your ansible playbook, so that it is copied to the config folder on the target.
+>    The file should contain the following content:
+>
+>    {% raw %}
+>    ```diff
+>    --- /dev/null
+>    +++ files/galaxy/config/tpv_rules_local.yml
+>    @@ -0,0 +1,30 @@
+>    +tools:
+>    +  testing:
+>    +    cores: 2
+>    +    mem: cores * 4
+>    +
+>    +destinations:
+>    +  local_env:
+>    +    runner: local_runner
+>    +    max_accepted_cores: 1
+>    +    params:
+>    +      tmp_dir: true
+>    +  singularity:
+>    +    runner: local_runner
+>    +    max_accepted_cores: 1
+>    +    params:
+>    +      singularity_enabled: true
+>    +    env:
+>    +      # Ensuring a consistent collation environment is good for reproducibility.
+>    +      LC_ALL: C
+>    +      # The cache directory holds the docker containers that get converted
+>    +      SINGULARITY_CACHEDIR: /tmp/singularity
+>    +      # Singularity uses a temporary directory to build the squashfs filesystem
+>    +      SINGULARITY_TMPDIR: /tmp
+>    +  slurm:
+>    +    inherits: singularity
+>    +    runner: slurm
+>    +    max_accepted_cores: 16
+>    +    params:
+>    +      native_specification: --nodes=1 --ntasks=1 --cpus-per-task={cores}
+>    +
+>    {% endraw %}
+>    ```
+>    {: data-commit="Add TPV local rules"}
+>
+>    In this TPV config, we have specified that the testing tool should use `2` cores. Memory has been defined as an expression and should be 4 times as much as cores, which, in this case, is 8GB.
+>    Note that the tool id is matched via a regular expression against the full tool id. For example, a full tool id for hisat may look like: `toolshed.g2.bx.psu.edu/repos/iuc/hisat2/hisat2/2.1.0+galaxy7`
```

Comment:
Can only `hisat2` be specified here as well? In which case any version of it would be matched? If so, would be nice to explicitly clarify this.